frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -28,27 +29,31 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   dirty,
   namespace,
   projects,
-}) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
-    <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection namespace={namespace} />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    <FormFooter
-      handleReset={handleReset}
-      errorMessage={status && status.submitError}
-      isSubmitting={isSubmitting}
-      submitLabel="Create"
-      sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
-      resetLabel="Cancel"
-    />
-  </Form>
-);
+}) => {
+  return (
+    <Form className="co-deploy-image" onSubmit={handleSubmit}>
+      <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
+      {values.type === EventSources.CronJobSource && <CronJobSection />}
+      {values.type === EventSources.SinkBinding && <SinkBindingSection />}
+      {values.type === EventSources.ApiServerSource && <ApiServerSection namespace={namespace} />}
+      {values.type === EventSources.CamelSource && <CamelSection />}
```

Comment:
You don't need to check specifically for `CamelSource`. Just create a white list of all the event sources that have their own section, if the current event source type is in that list render their respective sections else just render default `YAMLSection`.

Reply:
I would even suggest to move all the logic of checking the event sources into a simple util function `renderEventSoiurceSection` that has switch case. The default case of that switch returns `YAMLSection`.

Reply:
done